CMakeLists.txt

```
@@ -0,0 +1,202 @@
+    /WX # All warnings as error
+    /we4013 # Error warning C4013: 'function' undefined; assuming extern returning int
+    /we4700 # Error warning C4700: uninitialized local variable
+    /we4715 # not all control paths return a value
+    /we4003 # not enough actual parameters for macro
+    /wd4456 # disable warning C4456: declaration of 'xxx' hides previous local declaration
+  >
+  $<$<NOT:$<C_COMPILER_ID:MSVC>>:
+    -Wall
+    -Wextra
+    -Wpedantic
+  >
+)
+
+target_compile_definitions(base64 PRIVATE
+  $<$<C_COMPILER_ID:MSVC>:
+    # remove unnecessary warnings about unchecked iterators
+    _SCL_SECURE_NO_WARNINGS
+  >
+)
+
+########################################################################
+# SIMD settings
+include(TargetSIMDInstructionSet)
+define_SIMD_compile_flags()
+
+if (_TARGET_ARCH STREQUAL "x86" OR _TARGET_ARCH STREQUAL "x64")
+    macro(configure_codec _TYPE)
+        if (BASE64_WITH_${_TYPE})
+            message(STATUS "Add codec: lib/arch/${_DIR}/codec.c")
+            string(TOLOWER "${_TYPE}" _DIR)
+            set_source_files_properties("lib/arch/${_DIR}/codec.c" PROPERTIES
+                COMPILE_FLAGS "${COMPILE_FLAGS_${_TYPE}}"
+            )
+
+            if (${ARGC} GREATER 1 AND MSVC)
+                set_source_files_properties("lib/arch/${_DIR}/codec.c" PROPERTIES
+                    COMPILE_DEFINITIONS ${ARGV1}
+                )
+            endif()
+        endif()
+    endmacro()
+
+    configure_codec(SSSE3 __SSSE3__)
+    configure_codec(SSE41 __SSSE4_1__)
+    configure_codec(SSE42 __SSSE4_2__)
+    configure_codec(AVX)
+    configure_codec(AVX2)
+
+elseif (_TARGET_ARCH STREQUAL "arm")
+    set(BASE64_NEON32_CFLAGS "${COMPILE_FLAGS_NEON32}" CACHE STRING "the NEON32 compile flags (for 'lib/arch/neon32/codec.c')")
+    mark_as_advanced(BASE64_NEON32_CFLAGS)
+
+    if (BASE64_WITH_NEON32)
+        set_source_files_properties("lib/arch/neon32/codec.c" PROPERTIES
+            COMPILE_FLAGS "${BASE64_NEON32_CFLAGS} "
+        )
+    endif()
+
+#elseif (_TARGET_ARCH STREQUAL "arm64" AND BASE64_WITH_NEON64)
+
+endif()
+
+configure_file("${CMAKE_CURRENT_LIST_DIR}/cmake/config.h.in" "${CMAKE_CURRENT_BINARY_DIR}/config.h" @ONLY)
+target_include_directories(base64 PRIVATE "${CMAKE_CURRENT_BINARY_DIR}")
+
+########################################################################
+# OpenMP Settings
+if (BASE64_USE_OpenMP)
+    target_compile_options(base64
+        PRIVATE
+            ${OpenMP_C_FLAGS}
+    )
+endif()
+
+########################################################################
+if (BASE64_BUILD_TESTS)
+    enable_testing()
+    add_subdirectory(test)
+endif()
+
+########################################################################
+# install target
+if (BASE64_INSTALL_TARGET)
+    install(TARGETS base64 EXPORT base64-targets
+        RUNTIME DESTINATION bin/$<CONFIG>
+        LIBRARY DESTINATION lib/$<CONFIG>
+        ARCHIVE DESTINATION lib/$<CONFIG>
+        INCLUDES DESTINATION include
```

Better avoid stating those dirs manually, rather, use the `GNUInstallDirs` module: `include(GNUInstallDirs)`.  
It would then look like this:
```suggestion
        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
        INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
```

---

I included the `$<CONFIG>` part because of multi target generators like Visual Studio, but I no longer think that this matters much due to package managers and `CMakePresets.json`. Therefore I completely removed the path specification which is equivalent to your suggestion.